### PR TITLE
RHCLOUD-13104: Update signing tools to not remove comments

### PR DESCRIPTION
**Pull request by aleccohan**
Updated playbook-signature-tools to use the ruamel.yaml library as opposed to the oyaml library.  This will allow us to keep the header comments during the signing process while not disturbing the rest of the signing and verification logic.

Another update that occurred with this change is that the formatting of the template files slightly changed, the list sections marked by `-` will now not be indented due to the ruamel.yaml library's inability to support non-default yaml formatting.  This change to format will not effect the ability of the templates to be run by the Ansible runner.

JIRA: https://issues.redhat.com/browse/RHCLOUD-13104

### CODE REVIEW

**Comment by chenlizhong** [comment]
I tried this. Like Alec said the indents disappear. Ansible playbooks always have the indents. Can we use other tool to format it again after signing?

File: tools.py [hunk 1]
```diff
@@ -10,20 +9,18 @@
...
-class TemplateDumper(yaml.Dumper):
-    """
-    Template Dumper class:  Allows for correct YAML indentation
-    """
-    def increase_indent(self, flow=False, indentless=False):
-        return super(TemplateDumper, self).increase_indent(flow, False)
-
+yaml = YAML(typ='rt')
+yaml.default_flow_style = False
+yaml.width = 100
```
Comment (chenlizhong) [inline tools.py:24]: add `yaml.preserve_quotes = True` to keep quotes

File: Pipfile [hunk 2]
```diff
@@ -6,8 +6,8 @@ verify_ssl = true
 [dev-packages]
 
 [packages]
-pyyaml = "==5.4.1"
 python-gnupg = "==0.4.6"
+"ruamel.yaml" = "*"
```
Comment (SteveHNH) [inline Pipfile:10]: I would try to lock this one at a version. If for nothing else than it makes security happy.

File: tools.py [hunk 3]
```diff
@@ -92,7 +89,7 @@ def createSnippetHash(snippet):
         output: snippetHash (bytes)
     """
     snippetHash = hashlib.sha256()
-    serializedSnippet = bytes(yaml.dump(snippet, default_flow_style=False).encode("UTF-8"))
+    serializedSnippet = bytes(str(snippet).encode('UTF-8'))
```
Comment (SteveHNH) [inline tools.py:92]: You don't need to wrap these in bytes() because the encode() does that for you

**Review by dehort — APPROVED** [review]
LGTM!

**Pull request merged**